src/components/CardExpirationSection/CardExpirationSection.tsx

```diff
@@ -1,61 +1,51 @@
+  handleExpirationChange: (field: ExpirationKey, value: string) => void;
 };
-
-export default function CardExpirationSection({ expiration, setExpiration }: Props) {
-  const [expirationError, setExpirationError] = useState<string[]>(['', '']);
-
-  const handleExpirationChange = (index: number, value: string) => {
-    let errorMsg = '';
-    if (!/^[0-9]*$/.test(value)) {
-      errorMsg = '숫자만 입력 가능합니다.';
-      return;
-    }
-
-    if (index === 0) {
-      if (value !== '') {
-        const month = Number(value);
-        if (month < 1 || month > 12) {
-          errorMsg = '1부터 12 사이의 숫자를 입력해주세요.';
-        }
-      }
-    }
-    if (index === 1) {
-      if (value !== '' && value.length !== 2) {
-        errorMsg = '2자리 숫자를 입력해주세요.';
-      }
-    }
-
-    const updatedExpiration = [...expiration];
-    updatedExpiration[index] = value;
-    setExpiration(updatedExpiration);
-
-    const updatedError = [...expirationError];
-    updatedError[index] = errorMsg;
-    setExpirationError(updatedError);
-  };
+export default function CardExpirationSection({ expiration, expirationRef, handleExpirationChange }: CardExpirationSectionProps) {
+  const isError = expiration.month.errorMessage || expiration.year.errorMessage;
 
   return (
-    <div className={styles.sectionContainer}>
+    <div>
       <InputSection.TitleWrapper>
         <InputSection.Title title="카드 유효기간을 입력해 주세요" />
         <InputSection.SubTitle title="월/년도(MMYY)를 순서대로 입력해 주세요." />
       </InputSection.TitleWrapper>
-      <div className={styles.inputSection}>
+
+      <div>
         <InputSection.Label text="유효기간" />
-        <InputSection.InputWrapper
-          numbers={expiration}
-          onChange={handleExpirationChange}
-          valid={expirationError.map((msg) => msg === '')}
-          placeholders={['MM', 'YY']}
-          maxLength={2}
-        />
-        <div>{expirationError.map((msg, index) => msg && <InputSection.Error key={index} message={msg} />)}</div>
+
+        <InputSection.InputWrapper>
+          <Input
+            pattern="[0-9]{2}"
+            required
+            autoFocus={expiration.month.value == '' && expiration.year.value == ''}
+            value={expiration.month.value}
+            placeholder="MM"
+            isError={Boolean(expiration.month.errorMessage)}
+            onChange={(e) => handleExpirationChange('month', e.target.value)}
+            maxLength={2}
+            ref={expirationRef.month}
+          />
+          <Input
+            pattern="[0-9]{2}"
+            required
+            value={expiration.year.value}
+            placeholder="YY"
+            isError={Boolean(expiration.year.errorMessage)}
+            onChange={(e) => handleExpirationChange('year', e.target.value)}
+            maxLength={2}
+            ref={expirationRef.year}
+          />
+        </InputSection.InputWrapper>
+
+        {isError && <InputSection.Error message={expiration.month.errorMessage || expiration.year.errorMessage} />}
```

Comment:
```suggestion
        {isError && <InputSection.Error message={isError} />}
```
위에서 메타가 만든 `isError`를 재사용하거나 제가 말씀드린 방식을 사용한다면 `errorMessage`로 재사용할 수 있을 것 같아요🫡